### DISCUSSION TITLE
更新日志 - 22.03.6 - 20240430

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#1531] [Announcements]
# 系统
1. 修复了gnu版grep安装以后导致系统自动挂载的挂载点变成`-1`的问题，修复以后兼容busybox和gnu版grep。（例如安装iStore里面的“移动通信模块”会触发此bug） #1520
2. 其他一些使用grep和sed的脚本也兼容了busybox和gnu版

# 应用
* QuickStart（命令行界面，一般x86用户才会用上）：
  1. 允许修改DHCP客户端模式LAN口的IP（通常单网口机型才会默认LAN口DHCP客户端）。修改IP以后会变成静态地址模式，并且网关和DNS都不会配置，也就是无法上网，需要在网页后台进行额外配置，例如配置成旁路由
  2. 增加开放WAN口访问菜单，也就是允许WAN口访问后台，以方便开局配置或者PPPoE模式下强制允许WAN口访问
  3. 修复一级菜单选择与预期不符的bug

**Comment by tonyshva** [comment]
从iStoreOS 22.03.6 2024041213 升上来好像没法更新，
log:

Running Version: 'iStoreOS 22.03.6 2024041213'
Flashing Version: 'iStoreOS 22.03.6 2024043010'
Sending TERM to remaining processes ...
Sending signal TERM to S99adb-enablemo (14305)
Sending signal TERM to adb (14313)
Sending signal TERM to adb (14328)
Sending signal TERM to ddwebdav (14649)
Sending signal TERM to dockerd (14867)
Sending signal TERM to uuplugin (15396)
Sending signal TERM to uuplugin (15399)
Sending signal TERM to clash (23483)
Sending signal TERM to openclash_watch (24191)
Sending signal TERM to ddnstod (25791)
Sending signal TERM to sleep (4484)
Sending KILL to remaining processes ...
Switching to ramdisk...
Performing system upgrade...
Reading partition table from image...
Unable to find partition 1 device, skipped.
Unable to find partition 2 device, skipped.
Skip partition 3 >= 3 when upgrading
Writing new UUID to /dev/mmcblk1...
Upgrade completed

**Reply by jjm2473** [reply]
> Unable to find partition 1 device, skipped.

你的硬件是什么？开机以后终端执行`cat /tmp/.bootdisk; lsblk; ls -l /dev/`，然后开个issue把结果贴出来。

**Reply by Dragonborn2024** [reply]
MAJOR=179
MINOR=0
DEVNAME=mmcblk0
DEVTYPE=disk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
nbd0         43:0    0    0B  0 disk
nbd1         43:32   0    0B  0 disk
nbd2         43:64   0    0B  0 disk
nbd3         43:96   0    0B  0 disk
nbd4         43:128  0    0B  0 disk
nbd5         43:160  0    0B  0 disk
nbd6         43:192  0    0B  0 disk
nbd7         43:224  0    0B  0 disk
mmcblk0     179:0    0 59.5G  0 disk
├─mmcblk0p1 179:1    0  128M  0 part /boot
├─mmcblk0p2 179:2    0  256M  0 part /rom
├─mmcblk0p3 179:3    0    2G  0 part /overlay
└─mmcblk0p4 179:4    0   57G  0 part /mnt/mmc0-4
zram0       252:0    0    0B  0 disk
nbd8         43:256  0    0B  0 disk
nbd9         43:288  0    0B  0 disk
nbd10        43:320  0    0B  0 disk
nbd11        43:352  0    0B  0 disk
nbd12        43:384  0    0B  0 disk
nbd13        43:416  0    0B  0 disk
nbd14        43:448  0    0B  0 disk
nbd15        43:480  0    0B  0 disk
crw-------    1 root     root       10, 235 May 31 21:06 autofs
crw-------    1 root     root       10, 234 May 31 21:06 btrfs-control
drwxr-xr-x    3 root     root            60 May 31 21:06 bus
crw-------    1 root     root        5,   1 May 31 21:08 console
crw-------    1 root     root       10,  63 May 31 21:06 cpu_dma_latency
drwxr-xr-x    2 root     root           140 May 31 21:06 etherd
crw-rw-rw-    1 root     root        1,   7 May 31 21:06 full
crw-------    1 root     root       10, 229 May 31 21:06 fuse
crw-------    1 root     root       10, 183 May 31 21:06 hwrng
crw-------    1 root     root       89,   1 May 31 21:06 i2c-1
crw-------    1 root     root        1,  11 May 31 21:06 kmsg
srw-rw-rw-    1 root     root             0 May 31 21:06 log
crw-------    1 root     root       10, 237 May 31 21:06 loop-control
brw-------    1 root     root        7,   0 May 31 21:06 loop0
brw-------    1 root     root        7,   1 May 31 21:06 loop1
brw-------    1 root     root        7,   2 May 31 21:06 loop2
brw-------    1 root     root        7,   3 May 31 21:06 loop3
brw-------    1 root     root        7,   4 May 31 21:06 loop4
brw-------    1 root     root        7,   5 May 31 21:06 loop5
brw-------    1 root     root        7,   6 May 31 21:06 loop6
brw-------    1 root     root        7,   7 May 31 21:06 loop7
crw-------    1 root     root       10,  62 May 31 21:06 mISDNtimer
drwxr-xr-x    2 root     root            60 May 31 21:06 mapper
brw-------    1 root     root      179,   0 May 31 21:06 mmcblk0
brw-------    1 root     root      179,   1 May 31 21:06 mmcblk0p1
brw-------    1 root     root      179,   2 May 31 21:06 mmcblk0p2
brw-------    1 root     root      179,   3 May 31 21:06 mmcblk0p3
brw-------    1 root     root      179,   4 May 31 21:06 mmcblk0p4
brw-------    1 root     root       43,   0 May 31 21:06 nbd0
brw-------    1 root     root       43,  32 May 31 21:06 nbd1
brw-------    1 root     root       43, 320 May 31 21:06 nbd10
brw-------    1 root     root       43, 352 May 31 21:06 nbd11
brw-------    1 root     root       43, 384 May 31 21:06 nbd12
brw-------    1 root     root       43, 416 May 31 21:06 nbd13
brw-------    1 root     root       43, 448 May 31 21:06 nbd14
brw-------    1 root     root       43, 480 May 31 21:06 nbd15
brw-------    1 root     root       43,  64 May 31 21:06 nbd2
brw-------    1 root     root       43,  96 May 31 21:06 nbd3
brw-------    1 root     root       43, 128 May 31 21:06 nbd4
brw-------    1 root     root       43, 160 May 31 21:06 nbd5
brw-------    1 root     root       43, 192 May 31 21:06 nbd6
brw-------    1 root     root       43, 224 May 31 21:06 nbd7
brw-------    1 root     root       43, 256 May 31 21:06 nbd8
brw-------    1 root     root       43, 288 May 31 21:06 nbd9
drwxr-xr-x    2 root     root            60 May 31 21:06 net
crw-rw-rw-    1 root     root        1,   3 May 31 21:06 null
crw-------    1 root     root      108,   0 May 31 21:06 ppp
crw-rw-rw-    1 root     root        5,   2 May 31 21:09 ptmx
drwxr-xr-x    2 root     root             0 May 31 21:06 pts
crw-------    1 root     root        2,   0 May 31 21:06 ptyp0
crw-------    1 root     root        2,   1 May 31 21:06 ptyp1
crw-------    1 root     root        2,   2 May 31 21:06 ptyp2
crw-------    1 root     root        2,   3 May 31 21:06 ptyp3
crw-------    1 root     root        2,   4 May 31 21:06 ptyp4
crw-------    1 root     root        2,   5 May 31 21:06 ptyp5
crw-------    1 root     root        2,   6 May 31 21:06 ptyp6
crw-------    1 root     root        2,   7 May 31 21:06 ptyp7
crw-------    1 root     root        2,   8 May 31 21:06 ptyp8
crw-------    1 root     root        2,   9 May 31 21:06 ptyp9
crw-------    1 root     root        2,  10 May 31 21:06 ptypa
crw-------    1 root     root        2,  11 May 31 21:06 ptypb
crw-------    1 root     root        2,  12 May 31 21:06 ptypc
crw-------    1 root     root        2,  13 May 31 21:06 ptypd
crw-------    1 root     root        2,  14 May 31 21:06 ptype
crw-------    1 root     root        2,  15 May 31 21:06 ptypf
crw-rw-rw-    1 root     root        1,   8 May 31 21:06 random
crw-------    1 root     root      252,   0 May 31 21:06 rtc0
lrwxrwxrwx    1 root     root             8 May 31 21:06 shm -> /tmp/shm
drwxr-xr-x    2 root     root            60 May 31 21:06 snd
lrwxrwxrwx    1 root     root            15 May 31 21:06 stderr -> /proc/self/fd/2
lrwxrwxrwx    1 root     root            15 May 31 21:06 stdin -> /proc/self/fd/0
lrwxrwxrwx    1 root     root            15 May 31 21:06 stdout -> /proc/self/fd/1
crw-rw-rw-    1 root     root        5,   0 May 31 21:06 tty
crw-rw----    1 root     dialout     4,   0 May 31 21:06 tty0
crw-rw----    1 root     dialout     4,   1 May 31 21:06 tty1
crw-rw----    1 root     dialout     4,  10 May 31 21:06 tty10
crw-rw----    1 root     dialout     4,  11 May 31 21:06 tty11
crw-rw----    1 root     dialout     4,  12 May 31 21:06 tty12
crw-rw----    1 root     dialout     4,  13 May 31 21:06 tty13
crw-rw----    1 root     dialout     4,  14 May 31 21:06 tty14
crw-rw----    1 root     dialout     4,  15 May 31 21:06 tty15
crw-rw----    1 root     dialout     4,  16 May 31 21:06 tty16
crw-rw----    1 root     dialout     4,  17 May 31 21:06 tty17
crw-rw----    1 root     dialout     4,  18 May 31 21:06 tty18
crw-rw----    1 root     dialout     4,  19 May 31 21:06 tty19
crw-rw----    1 root     dialout     4,   2 May 31 21:06 tty2
crw-rw----    1 root     dialout     4,  20 May 31 21:06 tty20
crw-rw----    1 root     dialout     4,  21 May 31 21:06 tty21
crw-rw----    1 root     dialout     4,  22 May 31 21:06 tty22
crw-rw----    1 root     dialout     4,  23 May 31 21:06 tty23
crw-rw----    1 root     dialout     4,  24 May 31 21:06 tty24
crw-rw----    1 root     dialout     4,  25 May 31 21:06 tty25
crw-rw----    1 root     dialout     4,  26 May 31 21:06 tty26
crw-rw----    1 root     dialout     4,  27 May 31 21:06 tty27
crw-rw----    1 root     dialout     4,  28 May 31 21:06 tty28
crw-rw----    1 root     dialout     4,  29 May 31 21:06 tty29
crw-rw----    1 root     dialout     4,   3 May 31 21:06 tty3
crw-rw----    1 root     dialout     4,  30 May 31 21:06 tty30
crw-rw----    1 root     dialout     4,  31 May 31 21:06 tty31
crw-rw----    1 root     dialout     4,  32 May 31 21:06 tty32
crw-rw----    1 root     dialout     4,  33 May 31 21:06 tty33
crw-rw----    1 root     dialout     4,  34 May 31 21:06 tty34
crw-rw----    1 root     dialout     4,  35 May 31 21:06 tty35
crw-rw----    1 root     dialout     4,  36 May 31 21:06 tty36
crw-rw----    1 root     dialout     4,  37 May 31 21:06 tty37
crw-rw----    1 root     dialout     4,  38 May 31 21:06 tty38
crw-rw----    1 root     dialout     4,  39 May 31 21:06 tty39
crw-rw----    1 root     dialout     4,   4 May 31 21:06 tty4
crw-rw----    1 root     dialout     4,  40 May 31 21:06 tty40
crw-rw----    1 root     dialout     4,  41 May 31 21:06 tty41
crw-rw----    1 root     dialout     4,  42 May 31 21:06 tty42
crw-rw----    1 root     dialout     4,  43 May 31 21:06 tty43
crw-rw----    1 root     dialout     4,  44 May 31 21:06 tty44
crw-rw----    1 root     dialout     4,  45 May 31 21:06 tty45
crw-rw----    1 root     dialout     4,  46 May 31 21:06 tty46
crw-rw----    1 root     dialout     4,  47 May 31 21:06 tty47
crw-rw----    1 root     dialout     4,  48 May 31 21:06 tty48
crw-rw----    1 root     dialout     4,  49 May 31 21:06 tty49
crw-rw----    1 root     dialout     4,   5 May 31 21:06 tty5
crw-rw----    1 root     dialout     4,  50 May 31 21:06 tty50
crw-rw----    1 root     dialout     4,  51 May 31 21:06 tty51
crw-rw----    1 root     dialout     4,  52 May 31 21:06 tty52
crw-rw----    1 root     dialout     4,  53 May 31 21:06 tty53
crw-rw----    1 root     dialout     4,  54 May 31 21:06 tty54
crw-rw----    1 root     dialout     4,  55 May 31 21:06 tty55
crw-rw----    1 root     dialout     4,  56 May 31 21:06 tty56
crw-rw----    1 root     dialout     4,  57 May 31 21:06 tty57
crw-rw----    1 root     dialout     4,  58 May 31 21:06 tty58
crw-rw----    1 root     dialout     4,  59 May 31 21:06 tty59
crw-rw----    1 root     dialout     4,   6 May 31 21:06 tty6
crw-rw----    1 root     dialout     4,  60 May 31 21:06 tty60
crw-rw----    1 root     dialout     4,  61 May 31 21:06 tty61
crw-rw----    1 root     dialout     4,  62 May 31 21:06 tty62
crw-rw----    1 root     dialout     4,  63 May 31 21:06 tty63
crw-rw----    1 root     dialout     4,   7 May 31 21:06 tty7
crw-rw----    1 root     dialout     4,   8 May 31 21:06 tty8
crw-rw----    1 root     dialout     4,   9 May 31 21:06 tty9
crw-rw----    1 root     dialout     4,  64 May 31 21:06 ttyS0
crw-rw----    1 root     dialout     4,  65 May 31 21:06 ttyS1
crw-rw----    1 root     dialout     4,  74 May 31 21:06 ttyS10
crw-rw----    1 root     dialout     4,  75 May 31 21:06 ttyS11
crw-rw----    1 root     dialout     4,  76 May 31 21:06 ttyS12
crw-rw----    1 root     dialout     4,  77 May 31 21:06 ttyS13
crw-rw----    1 root     dialout     4,  78 May 31 21:06 ttyS14
crw-rw----    1 root     dialout     4,  79 May 31 21:06 ttyS15
crw-rw----    1 root     dialout     4,  66 May 31 21:06 ttyS2
crw-rw----    1 root     dialout     4,  67 May 31 21:06 ttyS3
crw-rw----    1 root     dialout     4,  68 May 31 21:06 ttyS4
crw-rw----    1 root     dialout     4,  69 May 31 21:06 ttyS5
crw-rw----    1 root     dialout     4,  70 May 31 21:06 ttyS6
crw-rw----    1 root     dialout     4,  71 May 31 21:06 ttyS7
crw-rw----    1 root     dialout     4,  72 May 31 21:06 ttyS8
crw-rw----    1 root     dialout     4,  73 May 31 21:06 ttyS9
crw-rw----    1 root     dialout     3,   0 May 31 21:06 ttyp0
crw-rw----    1 root     dialout     3,   1 May 31 21:06 ttyp1
crw-rw----    1 root     dialout     3,   2 May 31 21:06 ttyp2
crw-rw----    1 root     dialout     3,   3 May 31 21:06 ttyp3
crw-rw----    1 root     dialout     3,   4 May 31 21:06 ttyp4
crw-rw----    1 root     dialout     3,   5 May 31 21:06 ttyp5
crw-rw----    1 root     dialout     3,   6 May 31 21:06 ttyp6
crw-rw----    1 root     dialout     3,   7 May 31 21:06 ttyp7
crw-rw----    1 root     dialout     3,   8 May 31 21:06 ttyp8
crw-rw----    1 root     dialout     3,   9 May 31 21:06 ttyp9
crw-rw----    1 root     dialout     3,  10 May 31 21:06 ttypa
crw-rw----    1 root     dialout     3,  11 May 31 21:06 ttypb
crw-rw----    1 root     dialout     3,  12 May 31 21:06 ttypc
crw-rw----    1 root     dialout     3,  13 May 31 21:06 ttypd
crw-rw----    1 root     dialout     3,  14 May 31 21:06 ttype
crw-rw----    1 root     dialout     3,  15 May 31 21:06 ttypf
crw-rw-rw-    1 root     root        1,   9 May 31 21:06 urandom
crw-------    1 root     root        7,   0 May 31 21:06 vcs
crw-------    1 root     root        7,   1 May 31 21:06 vcs1
crw-------    1 root     root        7, 128 May 31 21:06 vcsa
crw-------    1 root     root        7, 129 May 31 21:06 vcsa1
crw-------    1 root     root        7,  64 May 31 21:06 vcsu
crw-------    1 root     root        7,  65 May 31 21:06 vcsu1
crw-rw-rw-    1 root     root        1,   5 May 31 21:06 zero
brw-------    1 root     root      252,   0 May 31 21:06 zram0
